### ISSUE TITLE
NO_PROXY

### ISSUE THREAD

**Issue by thekid** [xp-framework/http#15]
The `NO_PROXY` environment variable is not taken into account (while `no_proxy` is). Some programs expect this to be in all-caps, see https://wiki.archlinux.org/index.php/Proxy_settings.

See also https://www.npmjs.com/package/proxy-from-env and https://www.gnu.org/software/wget/manual/wget.html#Proxies

/cc @kiesel

**Comment by thekid** [comment]
On Windows/Cygwin, this is not a problem as environment variables are case-preserving, but case-insensitive. 

```sh
C:\tools\cygwin\home\Timm>set USER=Timm
C:\tools\cygwin\home\Timm>.\bin\xp.exe -w "getenv('USER')"
Timm
C:\tools\cygwin\home\Timm>.\bin\xp.exe -w "getenv('user')"
Timm
```

Not so on Mac/Unix:

```sh
vagrant@vagrant-ubuntu-vivid-64:~$ xp -w 'getenv("USER")'
vagrant
vagrant@vagrant-ubuntu-vivid-64:~$ xp -w 'getenv("user")'
false
```

**Comment by thekid** [comment]
:shipit: Released in [7.0.1](https://github.com/xp-framework/http/releases/tag/v7.0.1)
